### PR TITLE
[TS] LPS-161706 Add RedirectEntry's View permission on User Personal Site for Power Use

### PR DESCRIPTION
**Pull request by tototrinh**
Hi @adolfopa,
Please take a look at this PR and leave your comments. Thanks.
Ticket: https://issues.liferay.com/browse/LPS-161706
Previous PR: https://github.com/liferay-lima/liferay-portal/pull/3492

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
@tototrinh Can you write an integration test for this?

**Comment by tototrinh** [comment]
@adolfopa 
> @tototrinh Can you write an integration test for this?

Yes. Should the integration test be updated in this PR or in another PR? Thanks.

**Comment by adolfopa** [comment]
Thanks @tototrinh . If it is possible, it would be better to include the tests in this pr (in its own commit).

**Comment by tototrinh** [comment]
Hi @adolfopa,
I update the integration test. Please help take a look. Thanks.

File: modules/apps/redirect/redirect-test/src/testIntegration/java/com/liferay/redirect/web/internal/instance/lifecycle/test/RedirectEntryPermissionPortalInstanceLifecycleListenerTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author tototrinh
+ */
+@RunWith(Arquillian.class)
+public class RedirectEntryPermissionPortalInstanceLifecycleListenerTest {
```
Comment (adolfopa) [inline modules/apps/redirect/redirect-test/src/testIntegration/java/com/liferay/redirect/web/internal/instance/lifecycle/test/RedirectEntryPermissionPortalInstanceLifecycleListenerTest.java:45]: @tototrinh To verify that the upgrade process is indeed not needed, you'll need to add an additional test:
1. Disable the instance initializer you created.
2. Add a new company.
3. Enable the instance initializer component.
4. Check that the company you created in 2 has the correct permissions.

To see how to do that, take a look at the AMThumbnailsOSGiCommandsTest. There the ServiceComponentRuntime is used to disable/enable components programatically.

If the test doesn't pass, you'll need to create an upgrade process for existing companies.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/redirect/redirect-test/src/testIntegration/java/com/liferay/redirect/web/internal/instance/lifecycle/test/RedirectEntryPermissionPortalInstanceLifecycleListenerTest.java (hunk 1 above)
Comment (tototrinh) [reply modules/apps/redirect/redirect-test/src/testIntegration/java/com/liferay/redirect/web/internal/instance/lifecycle/test/RedirectEntryPermissionPortalInstanceLifecycleListenerTest.java:50]: Hi @adolfopa,
Thanks for your feedback. I update the code for test integration. Please help review it. Thanks.

File: modules/apps/redirect/redirect-test/src/testIntegration/java/com/liferay/redirect/web/internal/instance/lifecycle/test/RedirectEntryPermissionPortalInstanceLifecycleListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+		_testCompany = CompanyTestUtil.addCompany();
+
+		Assert.assertFalse(
+			_hasViewRedirectEntryPermission(_testCompany.getCompanyId()));
+
+		_enablePortalInstanceLifecycleListener(componentDescriptionDTO);
+
+		Assert.assertTrue(
+			_hasViewRedirectEntryPermission(_testCompany.getCompanyId()));
```
Comment (adolfopa) [inline modules/apps/redirect/redirect-test/src/testIntegration/java/com/liferay/redirect/web/internal/instance/lifecycle/test/RedirectEntryPermissionPortalInstanceLifecycleListenerTest.java:77]: When changing the state of the server in integration tests, you need to make sure you bring it back to its original state even in the case of failure. In this case, if an assertion fails, or some other error occurs before the _enable* method is executed, you'll leave the component disabled.
In this case, the fix is to wrap the tests in a try/finally block:
```
try {
    ...
    _disablePortalInstanceLifecycleListener(...);
    ...
}
finally {
    _enablePortalInstanceLifecycleListener(...);
}
```
Can you apply this change?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/redirect/redirect-test/src/testIntegration/java/com/liferay/redirect/web/internal/instance/lifecycle/test/RedirectEntryPermissionPortalInstanceLifecycleListenerTest.java (hunk 2 above)
Comment (tototrinh) [reply modules/apps/redirect/redirect-test/src/testIntegration/java/com/liferay/redirect/web/internal/instance/lifecycle/test/RedirectEntryPermissionPortalInstanceLifecycleListenerTest.java:77]: My bad, I will update the code. Thanks.

**Comment by adolfopa** [comment]
ci:forward

**Comment by adolfopa** [comment]
@tototrinh Can you fix the sf failure?

**Comment by tototrinh** [comment]
ci:test:sf

**Comment by tototrinh** [comment]
@adolfopa sorry for my bad. I fixed the SF issue. Thanks.

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**